ipq40xx: add e2600ac c1/c2 to DSA

zhangsd20 opened the pull request:
1.Convert E2600ac c1/c2 to DSA and enable it.
2.Currently, e2600ac-c1 cannot be built as the kernel is larger than the defined KERNEL_SIZE,
however, there is no bootloader limit for the kernel size so remove KERNEL_SIZE completely.

Signed-off-by: 张 鹏 <sd20@qxwlan.com>

zhangsd20 committed:
ipq40xx: add e2600ac c1 to dsa

Convert E2600ac c1 to DSA and enable it.

Signed-off-by: 张 鹏 <sd20@qxwlan.com>
[ rename port to more generic name ]
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

zhangsd20 committed:
ipq40xx: add e2600ac c2 to dsa

Convert E2600ac c2 to DSA and enable it.

Signed-off-by: 张 鹏 <sd20@qxwlan.com>
[ rename port to more generic name ]
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

robimarko commented on target/linux/ipq40xx/image/generic.mk:
```diff
@@ -1039,7 +1039,7 @@ define Device/qxwlan_e2600ac-c1
 	BOARD_NAME := e2600ac-c1
 	SOC := qcom-ipq4019
-	KERNEL_SIZE := 4096k
+	KERNEL_SIZE := 5120k
```
Is there a bootloader limit for the kernel size and is there one?

Cause, I see that FW is just stored in a single partition

robimarko replied:
Also, you forgot to remove `DEFAULT := n` so that after DSA conversion images get built again, I assume it was set due to small kernel size

zhangsd20 replied:
When I compiled, I found that the kernel size exceeded 4096k, resulting in a compilation error and I could not generate img firmware.

zhangsd20 replied:
Sorry I didn't pay attention to this DEFAULT := n

robimarko replied:
That doesn't answer my question, what is the bootloader limit for kernel size, or is there even one?
Cause, if there is none there is no point in defining the kernel limit, and if there is one just set it now and not update every time building breaks.

zhangsd20 replied:
With the default config alone, the kernel size exceeds 4019k.

Must be larger than 4019k to compile the simplest firmware.
So I expanded it to 5120k and there is enough space left.

If you try to compile the e2600ac c1, you will know why I did it

zhangsd20 replied:
no limit

robimarko replied:
Then remove the kernel size completely with that explanation as a dynamic splitter is being used anyway, which avoids the need to change the kernel size completely.

zhangsd20 replied:
Thanks!
ok, i will test it.
If it is fine,i will add a new commit and push it.

robimarko commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-e2600ac-c1.dts:
```diff
@@ -102,3 +110,27 @@
+&swport5 {
+	status = "okay";
+
+	label = "eth1";
```
What is the logic behind one port being called `lan1` and one `eth1`?

zhangsd20 replied:
It's just for us to use

robimarko replied:
Since you pointed out that there is silkscreen, name both ports according to it

robimarko commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-e2600ac-c2.dts:
```diff
@@ -137,3 +145,35 @@
+&swport5 {
+	status = "okay";
+
+	label = "eth1";
```
Same question, why `eth1`?

zhangsd20 replied:
ok,for ease of use,so it is eth1

robimarko replied:
Is there anything printed on the board or case with port names?

Cause, if not then `eth1` makes no sense, either make it `wan` or `lan2`

zhangsd20 replied:
yes, There is a silk screen on the housing of our equipment(eth0 eth1)

robimarko replied:
Great, then just label all ports to match the silkscreen instead of mix and matching

robimarko replied:
Since you pointed out that there is silkscreen, name both ports according to it

zhangsd20 commented:
When I compiled, I found that the kernel size exceeded 4096k, resulting in a compilation error and I could not generate img firmware.













在2023年02月22 23时23分，"Robert ***@***.***>写道：


@robimarko commented on this pull request.
In target/linux/ipq40xx/image/generic.mk:
> @@ -1039,7 +1039,7 @@ define Device/qxwlan_e2600ac-c1
 	DEVICE_VARIANT := C1
 	BOARD_NAME := e2600ac-c1
 	SOC := qcom-ipq4019
-	KERNEL_SIZE := 4096k
+	KERNEL_SIZE := 5120kIs there a bootloader limit for the kernel size and is there one?
Cause, I see that FW is just stored in a single partition
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

zhangsd20 commented:
ok,for ease of use,so it is eth1




在2023年02月22 23时32分，"Robert ***@***.***>写道：


@robimarko commented on this pull request.
In target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-e2600ac-c2.dts:
> +	nvmem-cell-names = "mac-address";
+	nvmem-cells = <&macaddr_gmac0>;
+};
+
+&swport4 {
+	status = "okay";
+	label = "lan2";
+
+	nvmem-cell-names = "mac-address";
+	nvmem-cells = <&macaddr_gmac0>;
+};
+
+&swport5 {
+	status = "okay";
+
+	label = "eth1";Same question, why eth1?
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

zhangsd20 commented:
It's just for us to use






在2023年02月22 23时31分，"Robert ***@***.***>写道：


@robimarko commented on this pull request.
In target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-e2600ac-c1.dts:
> +&switch {
+	status = "okay";
+};
+
+&swport4 {
+	status = "okay";
+	label = "lan1";
+
+	nvmem-cell-names = "mac-address";
+	nvmem-cells = <&macaddr_gmac0>;
+};
+
+&swport5 {
+	status = "okay";
+
+	label = "eth1";What is the logic behind one port being called lan1 and one eth1?
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

robimarko commented:
Please remove the first commit, there is no need for the enlargement commit

zhangsd20 committed:
ipq40xx: e2600ac-c1 remove KERNEL_SIZE

Currently, e2600ac-c1 cannot be built as the kernel is larger than the defined KERNEL_SIZE,
however, there is no bootloader limit for the kernel size so remove KERNEL_SIZE completely.

Signed-off-by: 张 鹏 <sd20@qxwlan.com>
[ improve commit title, fix merge conflict ]
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

zhangsd20 commented:
ok, I've removed it.










在2023年02月24 05时10分，"Robert ***@***.***>写道：


Please remove the first commit, there is no need for the enlargement commit
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

robimarko commented:
Rename the ports and put the KERNEL_SIZE patch first

robimarko commented:
Also, add whitespace after `ipq40xx:` in `ipq40xx:e2600ac-c1 remove KERNEL_SIZE` to pass CI formal checks

Ansuel commented:
@robimarko considering @zhangsd20 disappeared and this having little changes required...

From online pic I can see they really sell wifi motherboard... no enclosure and from the pic i can't see info on the silkscreen... I would rename the ports to sw-eth1..2..3 or other generic name. What do you think?

Or if we want just lan1..2..3 and wan... (considering we are assigning one port to wan)

zhangsd20 commented:
Yes.you are right.
I'm also thinking about this, but I've been busy lately and haven't had time to deal with it









在2023年04月13 01时08分，"Christian ***@***.***>写道：


@robimarko considering @zhangsd20 disappeared and this having little changes required...
From online pic I can see they really sell wifi motherboard... no enclosure and from the pic i can't see info on the silkscreen... I would rename the ports to sw-eth1..2..3 or other generic name. What do you think?
Or if we want just lan1..2..3 and wan... (considering we are assigning one port to wan)
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>

Ansuel commented:
@zhangsd20 I updated this to have generic name. If you are ok with these change, I would merge this pr. Waiting for your feedback.

zhangsd20 commented:
ok,I think your approach is right

zhangsd20 commented:
Hi,
Sorry, I've been busy with other things lately and don't have time to follow my pull request.
I don't see the modified content merged into any branch, 
what other changes do I need to make before it will be merged?

Thanks!

zhangsd20 commented:
Why I still don't see being merged in openwrt 22.03 ?

Neustradamus commented:
Replaces:
- https://github.com/openwrt/openwrt/pull/12044
